src/robotFactory.js

```diff
@@ -1,17 +1,73 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
+BaseRobot.prototype = {
+  goForward(n = 1) {
+    this.coords.y += n;
+  },
+  goBack(n = 1) {
+    this.coords.y -= n;
+  },
+  goRight(n = 1) {
+    this.coords.x += n;
+  },
+  goLeft(n = 1) {
+    this.coords.x -= n;
+  },
+  getInfo() {
+    return `Robot: ${this.name}, `
+      + `Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`;
+  },
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
+  this.coords.z = 0;
 }
 
-function DeliveryDrone() {
-  // implement
+FlyingRobot.prototype = BaseRobot.prototype;
+
+FlyingRobot.prototype.goUp = function f(n = 1) {
+  this.coords.z += n;
+};
+
+FlyingRobot.prototype.goDown = function f(n = 1) {
+  this.coords.z -= n;
+};
+
+function DeliveryDrone(name, weight, coords,
+  chipVersion, maxLoadWeight, currentLoad = null) {
```

Comment:
```suggestion
function DeliveryDrone(
  name, 
  weight, 
  coords,
  chipVersion, 
  maxLoadWeight, 
  currentLoad = null,
  ) {
```
one line for one argument